File: 6차_과제/7-1/c_src/7-1.c
#include <stdio.h>
#include "7-1.h"

int main() {
    treeNode* node7 = treemake('A', NULL, NULL);
    treeNode* node6 = treemake('B', NULL, NULL);
    treeNode* node5 = treemake('C', NULL, NULL);
    treeNode* node4 = treemake('D', NULL, NULL);
    treeNode* node3 = treemake('*', node7, node6);
    treeNode* node2 = treemake('/', node5, node4);
    treeNode* node1 = treemake('-', node3, node2);

    preorder(node1);
    printf("\n");
    
    inorder(node1);
    printf("\n");
    
    postorder(node1);
    printf("\n");

    return 0;
}

File: DataStructure/6차_과제/7-1/py_src/7_1.py
from header7_1 import treeNode
from func7_1_1 import *

def main():
    node7=treeMake('A', None, None)
    node6=treeMake('B', None, None)
    node5=treeMake('C', None, None)
    node4=treeMake('D', None, None)
    node3=treeMake('*', node7, node6)
    node2=treeMake('/', node5, node4)
    node1=treeMake('-', node3, node2)

    preorder(node1)
    print()

    inorder(node1)
    print()
    
    postorder(node1)
    print()

if __name__=="__main__":
    main()